src/RPCServer.js

```diff
@@ -21,8 +21,27 @@ class RPCServer {
     this._prefetchCount = prefetchCount
     this._timeoutMs = timeoutMs
 
-    this._callback = () => Promise.resolve()
     this._initializePromise = undefined
+
+    this.actions = new Map()
+  }
+
+  _callback (msg) {
+    let {action, data} = typeof msg === 'object' ? msg : {}
+    if (!this.actions.has(action)) {
+      return Promise.resolve()
+    }
+
+    let handler = this.actions.get(action)
+    return handler(data)
+  }
+
+  /**
+   * @param {string} action
+   * @param {Function} handler
+   */
+  registerAction (action, handler) {
+    this.actions.set(action, handler)
```

Comment:
Maybe we should handle scenarios where the action already has a handler.

Some ideas:
- throw error and keep original handler?
- create a 3rd parameter like: `override = false`?
- create a different method with `overrideAction`?

Then create tests for this scenario too.

Reply:
may I implement this features @tunderdomb?